Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM docker.io/airbyte/python-connector-base:2.0.0@sha256:c44839ba84406116e8ba68722a0f30e8f6e7056c726f447681bb9e9ece8bd916
```

Comment:
I don't see the connection. And I don't see any harm in publishing CDK "again". Since docker images (unlike PyPI versions) are immutable, we can always re-publish a versioned docker image with new security patches.

I agree this might not be the best "FROM" statement here, but I'm not following why a fundamentally different approach solves the core problem here. For instance, why not the base Python image be a product of this repo? Then bumping versions would happen naturally when either changes...

Reply:
@aaronsteers @ChristoGrab 
I think my statement will be better explained with a concrete example.

Let's say we install a new system dependency which is unrelated to a python CDK change or change some system settings.
If we are using the CDK version as a tag how do we, and consumers, know which tag got the update?

As a consumer you  have mulitple choices in terms of which tag to use:
- `:latest` it means **you don't care about the CDK version** and you're fine that the underlying image can change at any pull
- If you're using `:<cdk-version>` it means **you care about the CDK version** but you don't care about pulling an image which can be arbitrarily mutated. 
- If you're serious about build reproducibility you might use `:cdk-version@sha546:....` to pull a specific digest. But in this scenario, if you want to get the update we've made on the image you'd have to go to DockerHub and check if your sha differs from the latest advertised one.  In this scenario I would find it easier to publish a tag like `<cdk-version>-X`. It makes it explicit that we shipped an update to an image which did not change the CDK version. It's also easier to document changes in a changelog, instead of listing shas we would have proper version numbers.

So my suggestion is to push three tags:
* `:latest`: always mutable on each pull
* `:cdk-version`: static CDK version, mutable system deps on each pull
* `:cdk-version-<dockerfile-iteration-number>`: static cdk version, static system deps on each pull. Adding a sha digest to this tag would just be a security thing for consumers and not as a way to get reproductible builds.

Reply:
Addressed below!